### PR TITLE
Separate the game epoch from the solar system epoch

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: ksp_plugin/interface.cpp [hunk 1]
```diff
@@ -425,17 +429,14 @@ void principia__AdvanceTime(Plugin* const plugin,
...
 }
 
 void principia__ForgetAllHistoriesBefore(Plugin* const plugin,
                                          double const t) {
   journal::Method<journal::ForgetAllHistoriesBefore> m({plugin, t});
-  Instant const t0;
-  CHECK_NOTNULL(plugin)->ForgetAllHistoriesBefore(t0 + t * Second);
+  plugin->ForgetAllHistoriesBefore(FromGameTime(plugin, t));
   return m.Return();
 }
```
Comment (eggrobin) [inline ksp_plugin/interface.cpp]: You missed principia__CurrentTime.

File: ksp_plugin/interface_flight_plan.cpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ ksp_plugin::Burn FromInterfaceBurn(Plugin const* const plugin,
   return {burn.thrust_in_kilonewtons * Kilo(Newton),
```
Comment (eggrobin) [inline ksp_plugin/interface_flight_plan.cpp]: Plenty of Instant() left in this file. Perhaps make the constructor of Point private, befriending Frame and a J2000 function?

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 3]
```diff
@@ -1268,7 +1268,9 @@ where is_in_physics_bubble(vessel)
         ConfigNode initial_states =
             GameDatabase.Instance.GetConfigs(principia_initial_state_config_name)[0].config;
         plugin_ =
-            Interface.NewPlugin(double.Parse(initial_states.GetValue("epoch")),
+              //TODO(phl):Kompatibilitaet?
```
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: ä?

File: ksp_plugin/interface_flight_plan.cpp (hunk 2 above)
Comment (eggrobin) [inline ksp_plugin/interface_flight_plan.cpp]: At least, look for—and get rid of—all the Instant() in non-test code.

File: ksp_plugin/interface.cpp (hunk 1 above)
Comment (pleroy) [inline ksp_plugin/interface.cpp]: ... and a few others.

File: ksp_plugin/interface_flight_plan.cpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/interface_flight_plan.cpp]: Done for tests.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 3 above)
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: I thought there was a compatibility problem, but no.

File: physics/ephemeris_body.hpp [hunk 4]
```diff
@@ -958,7 +960,7 @@ void Ephemeris<Frame>::AppendMassiveBodiesState(
   CHECK(!trajectories_.empty());
   Instant const t_last_intermediate_state =
       checkpoints_.empty()
-          ? Instant() - std::numeric_limits<double>::infinity() * Second
+          ? J2000 - std::numeric_limits<double>::infinity() * Second
```
Comment (eggrobin) [inline physics/ephemeris_body.hpp]: #1096.

**Pull request merged**